ipie/addons/ephqmc/trial_wavefunction/coherent_state.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import scipy.linalg
+
+from ipie.addons.eph.trial_wavefunction.eph_trial_base import EphTrialWavefunctionBase
+from ipie.utils.backend import arraylib as xp
+
+#TODO greensfunctions are in estimators 
+class CoherentStateTrial(EphTrialWavefunctionBase):
+    """"""
+    def __init__(self, wavefunction, hamiltonian, num_elec, num_basis, verbose=False):
+        super().__init__(wavefunction, num_elec, num_basis, verbose=verbose)
+        self.num_elec = num_elec
+        self.nup, self.ndown = self.num_elec
+        self.w0 = hamiltonian.w0
+        self.m = hamiltonian.m
+        self.nsites = hamiltonian.nsites
+
+        self.beta_shift = np.squeeze(wavefunction[:, 0])
+        self.psia = wavefunction[:, 1:self.nup+1]
+        self.psib = wavefunction[:, self.nup+1:self.nup+self.ndown+1]
+
+    def calculate_energy(self, system, hamiltonian):
+        #TODO variational_energy_coherent_state in ipie.estimators.local_energy
+        ...
+
+    def calc_overlap(self, walkers) -> np.ndarray:
```

Comment:
All of the trial abstract methods should be pure methods. Do not modify the walker state.

Reply:
Ok scratch that. Currently only the overlap methods are pure. The green's function methods currently modify the walkers greens function. We should change this in the future.